### PR TITLE
Setup test app fs for each test

### PR DESCRIPTION
**Pull request by othercorey**
We can probably setup an `upgraded_apps` with the expected output for tests that change files. That comparison needs to be in TestCase though so will have to pass the function name to the assert.

### CODE REVIEW

File: tests/PHPUnitExtension.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Copy test app before each test.
+     *
+     * @return void
+     */
+    public function executeBeforeTest(string $test): void
+    {
+        [$className, $methodName] = explode('::', $test);
+        $className = substr($className, strrpos($className, '\\') + 1);
+        $testPath = TESTS . 'old_apps' . DS . $className . DS . $methodName;
```
Comment (markstory) [inline tests/PHPUnitExtension.php:33]: We could also have a single layer of directories and use `TESTS . 'old_apps' . DS . $className . '-' . $methodName` instead.

Comment (othercorey) [reply tests/PHPUnitExtension.php:33]: If that is preferred, I will change it.

Comment (markstory) [reply tests/PHPUnitExtension.php:33]: I don't have a strong preference. Was just thinking it would make a less deep file tree.

Comment (othercorey) [reply tests/PHPUnitExtension.php]: flatness has been restored.

**Pull request merged**